solutions/kubernetes/ceph-binary-build.sh

```diff
@@ -0,0 +1,299 @@
+
+   echo -e "\n\n########################################################################"
+   echo -e "# CEPH_REPO         : $CEPH_REPO                  "
+   echo -e "# CEPH_BRANCH       : $CEPH_BRANCH                "
+   echo -e "#########################################################################"
+}
+
+function prereq() {
+    add_primary_separator "\t\tPreparing Build node"
+
+    # Verify docker installation:
+    add_secondary_separator "Verify docker installation"
+    if ! which docker; then
+        add_secondary_separator "Please install Docker on Build Node Agent"
+        exit 1
+    fi
+}
+
+function build_ubuntu() {
+    add_primary_separator "Building ubuntu binary packages"
+
+    add_secondary_separator "Verfiy/pull Ubuntu docker image"
+    if [[ $(docker images | tail -n+2 | awk '{ print $1":"$2 }' | grep ubuntu) != "ubuntu:latest" ]]; then
+        docker pull ubuntu:latest
+    fi
+
+    add_secondary_separator "Make build directory and run Ubuntu container"
+    mkdir -p $MOUNT_LOCATION/ubuntu
+    docker run --rm -t -d --name ceph_ubuntu -v /$MOUNT_LOCATION/ubuntu:/home ubuntu:latest
+
+    pushd $MOUNT_LOCATION/ubuntu
+        add_common_separator "Removing previous files"
+        rm -rvf *
+    popd
+
+    add_secondary_separator "Write build script"
+    cp functions.sh $MOUNT_LOCATION/ubuntu
+    pushd $MOUNT_LOCATION/ubuntu
+        cat << EOF > build.sh
+#/bin/bash
+
+source /home/functions.sh
+
+add_common_separator "Update repolist cache and install prerequisites"
+apt update && apt install git -y
+DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends tzdata
+pushd /home/
+    add_common_separator "Clone Repo"
+    git clone $CEPH_REPO -b $CEPH_BRANCH
+
+    pushd ceph
+        add_common_separator "Checkout Submodules"
+        git submodule update --init --recursive
+
+        add_common_separator "Install Dependencies"
+        ./install-deps.sh
+
+        add_common_separator "Make Source Tarball"
+        ./make-dist
+        
+        mv ceph-*tar.bz2 ../
+        version=\$(git describe --long --match 'v*' | sed 's/^v//')
+    popd
+
+    tar -xf ceph-*tar.bz2    
+    pushd /home/ceph-"\$version"
+        add_common_separator "Start Build"
+        dpkg-buildpackage -us -uc
+    popd
+
+    add_common_separator "List generated binary packages (*.deb)"
+    ls *.deb
+popd
+EOF
+
+    chmod +x build.sh
+    popd
+
+    add_secondary_separator "Run build script"
+    docker exec ceph_ubuntu bash /home/build.sh
+
+    add_primary_separator "\t\tDestroy build environment"
+    docker stop ceph_ubuntu
+}
+
+function build_centos() {
+    add_primary_separator "Building centos binary packages"
+
+    add_secondary_separator "Verfiy/pull CentOS docker image"
+    if [[ $(docker images | tail -n+2 | awk '{ print $1":"$2 }' | grep centos) != "centos:latest" ]]; then
+        docker pull centos:latest
```

Comment:
Let's use some fixed version instead of the latest. Latest is a kind of moving target and may break.